src/main/kotlin/net/atos/zac/mail/model/Bronnen.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.zac.mail.model
+
+import net.atos.client.zgw.drc.model.generated.EnkelvoudigInformatieObject
+import net.atos.client.zgw.zrc.model.Zaak
+import org.flowable.task.api.TaskInfo
+
+class Bronnen private constructor(
+    val zaak: Zaak,
+    val document: EnkelvoudigInformatieObject?,
+    val taskInfo: TaskInfo?
+) {
+    class Builder {
+        private lateinit var zaak: Zaak
+        private var document: EnkelvoudigInformatieObject? = null
+        private var taskInfo: TaskInfo? = null
+
+        fun add(zaak: Zaak): Builder {
+            this.zaak = zaak
+            return this
+        }
+
+        fun add(document: EnkelvoudigInformatieObject): Builder {
+            this.document = document
+            return this
+        }
+
+        fun add(taskInfo: TaskInfo): Builder {
+            this.taskInfo = taskInfo
+            return this
+        }
+
+        fun build(): Bronnen {
+            return Bronnen(zaak, document, taskInfo)
+        }
+    }
+
+    companion object {
+        fun fromZaak(zaak: Zaak): Bronnen {
+            return Builder().add(zaak).build()
+        }
+    }
```

Comment:
No need for a companion object here? It's just a plain function. Better even probably to make it into an extension function? 


```suggestion
        fun fromZaak(zaak: Zaak): Bronnen {
            return Builder().add(zaak).build()
        }
```

Reply:
Java code uses this as static function, so I annotated it with `@JvmStatic`